### PR TITLE
feat(server,web): remove external path nonsense and make libraries admin-only

### PR DESCRIPTION
**Pull request by etnoy**
We make the library configuration admin-only. This removes a lot of complexity regarding external paths.

There is a new admin panel for external libraries. The configuration pane might be a better place for it, but it won't fit in a config file so I'm unsure.

User library settings have been removed.

image

We have a picker for library owner:
image

### CODE REVIEW

File: server/src/immich/controllers/library.controller.ts
Comment (jrasm91) [inline server/src/immich/controllers/library.controller.ts:36]: We definitely should not use a POST for a GET. Alternatively we could use a query param to differentiate between getting my libraries vs getting all the libraries.

File: server/e2e/api/specs/library.e2e-spec.ts
Comment (jrasm91) [inline server/e2e/api/specs/library.e2e-spec.ts:197]: Can you assert the body message is correct as well?

File: server/src/domain/library/library.service.ts
Comment (jrasm91) [inline server/src/domain/library/library.service.ts:194]: You should move this logic into the `requirePermission` implementation. It should return true if the user is an admin or execute the check otherwise. This is access control logic though, and that should exist inside of the access.core.ts file.

File: server/src/immich/controllers/library.controller.ts
Comment (jrasm91) [inline server/src/immich/controllers/library.controller.ts:38]: Why do you have admin checks inside of this route if it is already unreachable to non-admins because of this guard?

File: server/src/immich/controllers/library.controller.ts
Comment (etnoy) [reply server/src/immich/controllers/library.controller.ts]: I think I've reworked this now

File: server/e2e/api/specs/library.e2e-spec.ts
Comment (etnoy) [reply server/e2e/api/specs/library.e2e-spec.ts]: Done

File: server/src/domain/library/library.service.ts
Comment (etnoy) [reply server/src/domain/library/library.service.ts]: Done

File: server/src/immich/controllers/library.controller.ts
Comment (etnoy) [reply server/src/immich/controllers/library.controller.ts]: Done, we're only relying on the controller guard now

**Review by jrasm91 — APPROVED** [review]
The server code LGTM. This is a nice change and should remove a lot of confusion!

**Comment by alextran1502** [comment]
Can we split the documentation to a separate PR that will be merged after the next release?

**Comment by jrasm91** [comment]
That, or we do another release and merge this right before.

**Comment by etnoy** [comment]
I'll split the pr, that'll be the easiest

**Comment by etnoy** [comment]
Ready to merge, after the next release we merge #7526

**Comment by bo0tzz** [comment]
There are still some docs changes in this PR

File: docs/docs/features/libraries.md
Comment (alextran1502) [inline docs/docs/features/libraries.md:1]: We still need to remove this doc change from this PR

Comment (etnoy) [reply docs/docs/features/libraries.md:1]: Sorry, I think I have a forgotten push in there. Will fix when I'm back at the computer

**Comment by etnoy** [comment]
> There are still some docs changes in this PR

Fixed, had forgot to push

**Pull request merged**
